mod_test/controllers.py

```diff
@@ -144,6 +144,29 @@ def get_data_for_test(test, title=None):
     }
 
 
+@mod_test.route('/get_json_data/<test_id>')
+def get_json_data(test_id):
+    test = Test.query.filter(Test.id == test_id).first()
+    pr_data = test.progress_data()
+    progress_array = []
+    for entry in test.progress:
+        progress = {'timestamp': '', 'status': '', 'message': ''}
+        progress['timestamp'] = entry.timestamp.strftime(
+            '%Y-%m-%d %H:%M:%S (%Z)')
+        progress['status'] = entry.status.description
+        progress['message'] = entry.message
+        progress_array.append(progress)
```

Comment:
Why not simply do a progress_array.append({'timestamp': entry.timestamp.strftime('%Y-%m-%d %H:%M:%S (%Z)'), ...})

(but well-formatted)? Is just as legible :)